src/frontends/lean/decl_util.cpp

```diff
@@ -79,11 +107,15 @@ expr parse_single_header(parser & p, declaration_name_scope & scope,
         while (is_pi(it)) it = binding_body(it);
         expr const & C = get_app_fn(it);
         name ns = get_namespace(p.env());
-        if (is_constant(C) && !ns.is_anonymous()) {
-            c_name = const_name(C);
-            scope.set_name(c_name);
-        } else if (is_constant(C) && is_app(it) && is_constant(get_app_fn(app_arg(it)))) {
-            c_name = const_name(get_app_fn(app_arg(it))) + const_name(C);
+        if (is_constant(C) && is_app(it) && is_constant(get_app_fn(app_arg(it)))) {
+            /* See the note "Naming instances" above. */
+            name class_name = const_name(C);
+            name arg_name = const_name(get_app_fn(app_arg(it)));
+            if (is_prefix_of(ns, class_name))
+                class_name = class_name.replace_prefix(ns, name());
+            if (is_prefix_of(ns, arg_name))
+                arg_name = arg_name.replace_prefix(ns, name());
+	    c_name = arg_name + class_name;
```

Comment:
indentation

Reply:
Argh, thanks. I fixed my emacs to not use tabs.